ansible_collections/arista/avd/roles/eos_designs/templates/base/base.py

```diff
@@ -0,0 +1,559 @@
+                    event_handler[handler_name].update({'trigger': trigger})
+                if regex is not None:
+                    event_handler[handler_name].update({'regex': regex})
+            return event_handler
+        return None
+
+    @cached_property
+    def load_interval(self):
+        load_interval_default = get(self._hostvars, 'load_interval_default')
+        if load_interval_default is not None:
+            return {'default': load_interval_default}
+        return None
+
+    @cached_property
+    def queue_monitor_length(self):
+        queue_monitor_length = get(self._hostvars, 'queue_monitor_length')
+        if queue_monitor_length is not None:
+            queue_monitor_length_dict = {'enabled': True}
+            queue_monitor_length_notifying = get(queue_monitor_length, 'notifying')
+            feature_support = get(self._platform_settings, 'feature_support')
+            queue_monitor_length_notify = ""
+            if feature_support is not None:
+                queue_monitor_length_notify = get(feature_support, 'queue_monitor_length_notify')
+            if (
+                    queue_monitor_length_notifying is not None and
+                    queue_monitor_length_notify is not False
+            ):
+                queue_monitor_length_dict.update({'notifying': queue_monitor_length_notifying})
+            if get(queue_monitor_length, 'log') is not None:
+                queue_monitor_length_dict.update({'log': get(queue_monitor_length, 'log')})
+            return queue_monitor_length_dict
+        return None
+
+    @cached_property
+    def name_server(self):
+        name_servers = get(self._hostvars, 'name_servers')
+        if name_servers is not None:
+            return {
+                'source':
+                    {
+                        'vrf': self._mgmt_interface_vrf
+                    },
+                'nodes': name_servers
+            }
+        return None
+
+    @cached_property
+    def redundancy(self):
+        redundancy = get(self._hostvars, 'redundancy')
+        if redundancy is not None:
+            protocol = get(redundancy, 'protocol')
+            if protocol is not None:
+                return {'protocol': protocol}
+        return None
+
+    @cached_property
+    def snmp_server(self):
+        snmp_settings = get(self._hostvars, 'snmp_settings')
+        if snmp_settings is not None:
+            snmp_server = {}
+            compute_local_engineid = get(snmp_settings, 'compute_local_engineid')
+            if compute_local_engineid is True:
+                local_engine_id = sha1(f"{self._hostname}{self._mgmt_ip}".encode('utf-8')).hexdigest()
+                snmp_server.update({'engineid': {'local': local_engine_id}})
+            contact = get(snmp_settings, 'contact')
+            if contact is not None:
+                snmp_server.update({'contact': contact})
+            if get(snmp_settings, 'location') is not None:
```

Comment:
please refactor this to add the elements to a list and then use a list comprehension to check for `is not None` and finally use `" ".join(location_elements)`